src/utils.tsx

```diff
@@ -12,23 +13,16 @@ export type Percentage = number | `${number}` | `${number}%`;
 export type Ephemeral = number | `${number}` | `${number}em`;
 export const typeToUnit = { Pixel: "px", Percentage: "%", Ephemeral: "em" };
 
-export function mjmlReactComponentFactory<P>(
-  mjmlElementName: string
-): React.FC<P> {
-  return function ({ children, ...props }) {
-    return React.createElement(
-      mjmlElementName,
-      convertPropsToMjmlAttributes(props),
-      children
-    );
-  };
-}
+type MJMLDangerouslySetInnerHTML = {
+  __html: string;
+};
 
 export function convertPropsToMjmlAttributes<P>(props: {
   [K in keyof P]: unknown;
 }) {
   const mjmlProps = Object.entries(props).reduce((mjmlProps, [prop, value]) => {
-    const mjmlProp = kebabCase(prop);
+    const mjmlProp =
+      prop === DANGEROUSLY_SET_INNER_HTML ? prop : kebabCase(prop);
```

Comment:
It's outside the scope of this PR but I'm wondering if we should better support the default [React props](https://reactjs.org/docs/dom-elements.html#differences-in-attributes)

1. Only kabab mjml props (maybe pass a list of props into this function or the mjml-props for a specific component?)
2. Add additional props to the component types